tests/lib/date_time/src/main.c

```diff
@@ -275,6 +275,12 @@ static void test_date_time_teardown(void)
 
 void test_main(void)
 {
+	/* Delay to ensure that k_uptime_get returns positive non-zero value
+	 * irrespective of what CONFIG_SYS_CLOCK_TICKS_PER_SEC value is.
+	 * This has been an issue in QEMU tests.
+	 */
+	k_sleep(K_SECONDS(1));
```

Comment:
If this delay is considered a problem in tests that otherwise would take few hundred milliseconds, we could add a delay of  `1000/CONFIG_SYS_CLOCK_TICKS_PER_SEC` milliseconds whenever `CONFIG_SYS_CLOCK_TICKS_PER_SEC < 1000` and otherwise 1 millisecond.